### DISCUSSION TITLE
[Notification/message] they can not display on android 12

### DISCUSSION THREAD

**Discussion by Minhuun** [invertase/react-native-firebase#6379] [Q&A]
Hi all, my app worked well on SDK target 30 && below but when update SDK target 31 so notification & message did not display in system notification center, they was not dropdown at status bar.

Please, give me some solution fix this.

<meta-data android:name="com.google.firebase.messaging.default_notification_icon" android:resource="@drawable/notification_trans"/>
        <meta-data android:name="com.google.firebase.messaging.default_notification_channel_id" android:value="my_default_channel"/>
        <activity android:name="com.facebook.react.devsupport.DevSettingsActivity" android:exported="true"/>
        <service android:name="io.invertase.firebase.messaging.RNFirebaseMessagingService" android:exported="true">
            <intent-filter>
            <action android:name="com.google.firebase.MESSAGING_EVENT" />
            </intent-filter>
        </service>
        <service android:name="io.invertase.firebase.messaging.RNFirebaseInstanceIdService" android:exported="true">
            <intent-filter>
            <action android:name="com.google.firebase.INSTANC
<img width="1046" alt="Screen Shot 2022-07-11 at 15 41 42" src="https://user-images.githubusercontent.com/92594267/178224167-bcf42a51-fea6-425d-ba85-b473332b3297.png">
E_ID_EVENT"/>
            </intent-filter>
        </service>
        <service android:name="io.invertase.firebase.messaging.RNFirebaseBackgroundMessagingService" android:exported="true"/>

**Comment by Minhuun** [comment]
My firebase version: https://github.com/smartpayvn/react-native-firebase#v5.5.6.2

**Reply by Lunziyuan** [reply]
same with you V5.5.6

**Reply by mikehardy** [reply]
With apologies those versions will receive no support - not only are they from nearly 3 years ago, they aren't even maintained here. Did you notice that is a different repo?

Please upgrade to current versions from this repository if you want help troubleshooting, then we can take a look

**Comment by mikehardy** [comment]
With apologies, I am not sure at all what the problem is, or how to reproduce it
https://stackoverflow.com/help/how-to-ask
https://stackoverflow.com/help/minimal-reproducible-example

There just aren't enough details (versions, code, screenshots, etc) to figure it out

**Reply by yash-raj-py** [reply]
I am facing the same issue 
Now we have to give as  targetsdk 31 , notification don't not come for Android 12 and above devices .

**Comment by mikehardy** [comment]
react-native-permissions and notifee.app both include support for requesting the new Android 12 notification permission if I understand correctly. I use react-native-permissions to centralize all my permissions and it seems to work fine ?

**Reply by yash-raj-py** [reply]
So what change i should do 
I have react native firebase 5.4.0

**Reply by mikehardy** [reply]
@yash-raj-py I have no idea *what* I would do if I was using a version that old. I would stop using it immediately. I would upgrade to current stable so that I could ask questions here and receive support.

**Reply by yash-raj-py** [reply]
Ok , can u tell me one thing which gradle version and plugin in should upgrade my react native 0.59.10 so that i can install firebase versions.

**Reply by mikehardy** [reply]
There is no way at all that react-native 0.59 is supportable at this point. The lowest react-native that has a reasonable chance at support by the react-native team themselves is 0.68. I think react-native 0.65 has a chance at working with modern toolchains (java/gradle or Xcode) but I wouldn't use less 0.65 at all and I'd go right to 0.70 - https://react-native-community.github.io/upgrade-helper/

You can only defer maintenance so long before a car breaks down, as an analogy

**Comment by htran844** [comment]
hope my answer can help you
[when-receiving-a-push-message-on-android12-api-level-31-32-could-not-send-no](https://stackoverflow.com/questions/72985612/when-receiving-a-push-message-on-android12-api-level-31-32-could-not-send-no/74599005#74599005)

**Reply by yash-raj-py** [reply]
Which firebase messaging, iid and core version you have used

**Reply by yash-raj-py** [reply]
For implementation
 In app / gradle

**Reply by htran844** [reply]
I use
 `implementation 'com.google.firebase:firebase-core:16.0.0'
    implementation 'com.google.firebase:firebase-messaging:18.0.0'`
2 year ago, i am not change it

**Reply by yash-raj-py** [reply]
And this works for Android 12 ?

**Reply by htran844** [reply]
I am tested on android 12. this work. trust me

**Reply by yash-raj-py** [reply]
Can I have your app/gradle and gradle file ?
Ya i tried but dint work

**Reply by yash-raj-py** [reply]
And thank you so much for help

**Reply by htran844** [reply]
update in build.gradle/app :
 `dependencies {
    // implementation 'com.google.firebase:firebase-core:16.0.0'
    // implementation 'com.google.firebase:firebase-messaging:18.0.0'

    implementation "com.google.firebase:firebase-iid:21.1.0" 
    implementation "com.google.firebase:firebase-messaging:23.1.0"
    implementation "com.google.firebase:firebase-inappmessaging:20.2.0" 
  // ...
}

android {
  compileSdkVersion 31
  defaultConfig {
    targetSdkVersion 31
    // ...
  }
  // ...
}`

**Reply by yashpyraj** [reply]
we need inappmessaging and no core is required?

**Reply by htran844** [reply]
this work for me, test in samsung s21

**Reply by yashpyraj** [reply]
just for  my clarification can you send few files of your project , android/gradle , android/app/gradle and androidmenifest.xml ?

**Reply by yashpyraj** [reply]
can you also see after these changes it works android 11 or below version too .

**Reply by mikehardy** [reply]
IID is deprecated, and has been for nearly 2 years.
Core at v16 is from around 3 years go.

It seems quite odd to me to be so persistent in using old software. These are versions from before there even *was* a notification permission on Android.

**Reply by yashpyraj** [reply]
i have upgraded to firebase version v6 and what is happing i am getting token and notification for android 12 but on testing with android 11 the token dint came and showing error FIS_AUTH_ERROR
```
[FCMService] getToken rejected  NativeFirebaseError: [messaging/unknown] java.io.IOException: java.util.concurrent.ExecutionException: java.io.IOException: FIS_AUTH_ERROR
    at FirebaseMessagingModule.getToken (c22cefdd-e235-4c15-8fca-87f893bb2220:193482:28)
    at _callee3$ (c22cefdd-e235-4c15-8fca-87f893bb2220:189492:77)
    at tryCatch (c22cefdd-e235-4c15-8fca-87f893bb2220:34329:19)
    at Generator.invoke [as _invoke] (c22cefdd-e235-4c15-8fca-87f893bb2220:34466:24)
    at Generator.next (c22cefdd-e235-4c15-8fca-87f893bb2220:34362:23)
    at tryCatch (c22cefdd-e235-4c15-8fca-87f893bb2220:34329:19)
    at invoke (c22cefdd-e235-4c15-8fca-87f893bb2220:34387:22)
    at c22cefdd-e235-4c15-8fca-87f893bb2220:34412:13
    at tryCallTwo (c22cefdd-e235-4c15-8fca-87f893bb2220:38050:7)```

**Reply by mikehardy** [reply]
> firebase version v6 

We are on v16.4.6 here at the moment. "v6" is not really a specific version.

Your error has lots of root causes (https://search.brave.com/search?q=FIS_AUTH_ERROR) but first let's make sure you are actually on current stable versions. You will most likely want to download a new set of config files from the firebase web console (that is, the google services plist and json for your ios and android app - it's a common source of errors if you are upgrading from really really old versions)

**Reply by yashpyraj** [reply]
in package.json 
```
"@react-native-firebase/app": "^16.4.6",
    "@react-native-firebase/messaging": "^16.4.6",
```
import
```
import messaging from "@react-native-firebase/messaging";

```

fetching token 
```
const token = await messaging()
			.getToken()
			.catch((error) => {
				console.log("[FCMService] getToken rejected ", error);
			});
		console.log("firebase.messaging().getToken - Done", token);

		if (token !== null && token !== "" && token !== undefined) {
			console.log("sendFcmEventOnLogin", "executed");
			this.onChangeToken(token);
		}

```
i have updated google services json ( why this working for android 12 but not for android 11 )

**Reply by trunglemobiledev** [reply]
> 

Thanks i'm fixed with rn-firebase 5.6.0

**Reply by trunglemobiledev** [reply]
> hope my answer can help you [when-receiving-a-push-message-on-android12-api-level-31-32-could-not-send-no](https://stackoverflow.com/questions/72985612/when-receiving-a-push-message-on-android12-api-level-31-32-could-not-send-no/74599005#74599005)

i dont open app from notification when backgroud

**Reply by htran844** [reply]
> > hope my answer can help you [when-receiving-a-push-message-on-android12-api-level-31-32-could-not-send-no](https://stackoverflow.com/questions/72985612/when-receiving-a-push-message-on-android12-api-level-31-32-could-not-send-no/74599005#74599005)
> 
> i dont open app from notification when backgroud (Tôi không thể mở app từ thông báo trạng thái ngoài app đê đế chi tiết thông báo trong app)

check your code or find issue 'open notification from background'

**Reply by trunglemobiledev** [reply]
> > > hope my answer can help you [when-receiving-a-push-message-on-android12-api-level-31-32-could-not-send-no](https://stackoverflow.com/questions/72985612/when-receiving-a-push-message-on-android12-api-level-31-32-could-not-send-no/74599005#74599005)
> > 
> > 
> > i dont open app from notification when backgroud (Tôi không thể mở app từ thông báo trạng thái ngoài app đê đế chi tiết thông báo trong app)
> 
> check your code or find issue 'open notification from background'

Thanks i fixed !🥰

**Reply by thawzinlinn15** [reply]
> hope my answer can help you [when-receiving-a-push-message-on-android12-api-level-31-32-could-not-send-no](https://stackoverflow.com/questions/72985612/when-receiving-a-push-message-on-android12-api-level-31-32-could-not-send-no/74599005#74599005)

hello brother, I try with your solution, but only foreground noti worked, background noti didn't received on Android 12, could you help me :')

**Reply by thawzinlinn15** [reply]
> > hope my answer can help you [when-receiving-a-push-message-on-android12-api-level-31-32-could-not-send-no](https://stackoverflow.com/questions/72985612/when-receiving-a-push-message-on-android12-api-level-31-32-could-not-send-no/74599005#74599005)
> 
> hello brother, I try with your solution, but only foreground noti worked, background noti didn't received on Android 12, could you help me :')

react-native: 0.59.10. (my project not easy to update react-native version :') )
react-native-firebase: 5.6.0
compileSdkVersion         = 31
minSdkVersion             = 18
targetSdkVersion          = 31

**Reply by mikehardy** [reply]
react-native 0.59 is definitely no longer supported, easy to upgrade or not.
In general two major versions in the past are supported - similar to this policy https://github.com/reactwg/react-native-releases#releases-support-policy - with some effort made such that older versions *can* work, but 0.59 is definitely not in that range

**Reply by thawzinlinn15** [reply]
not easy to upgrade the react-native version, Is there any way to fix (background noti) that issue?
For foreground noti, I fixed in the node_modules.

**Reply by mikehardy** [reply]
> react-native 0.59 is definitely no longer supported, easy to upgrade or not.

This statement is authoritative, apologies
